1.sh

```diff
@@ -1,35 +1,34 @@
 #! /bin/sh
 
-if [-d $1] && [-d $2]
+if [ -d "$1" ] && [ -d "$2" ]
 then
 	echo "Dirs exist"
-elif [! -d $2] 
+elif [ ! -d "$2" ] 
 	echo "Flash path doesn't exist"
-	mkdir $2
+	mkdir "$2"
 fi
 
-for file in $1/*
+for file in "$1"/*
 do
-	echo $(basename $file)
-	flashfile=$($2/$(basename $file))
-	if [! -f $flashfile]; then
-		echo "Copying files to $2"
-		cp $file $2
+	echo "$(basename $file)"
+	flashfile="$2/$(basename $file)"
+
+	if [ ! -f "$flashfile" ]; then
+		echo "Copying files to '$2'"
+		cp "$file" "$2"
 	
 	elif [ $(ls -l $file) qt $(ls -l $flashfile) ]
-		echo "Flash version is newer, replace existing one? Y\N"
-
-	read reply
-	case $reply in
-	y|Y) echo "Copying now..."
-		cp $flashdrive $1
-	;;
-	n|N) echo "Staying as we were" 
-		exit 2
-	*) echo "Wrong input"
-	;;
-	esac
+		read -e -n 1 -p "Flash version is newer, replace existing one? [Y|N] "
```

Comment:
While not necessary, I would suggest this approach as it's more concise, in my opinion.

The bash builtin `read` is capable of providing a prompt to the user, by using the `-p` flag. See `help read` for further, albeit limited information. The `-e` flag allows for a new line to be displayed, where otherwise it wouldn't. The `-n` flag is for having read accept only the given number (in this case, `1`) of characters before immediately (without the need for the Enter or Return key to be pressed) proceeding.

Lastly, here, `[Y|N]` is a traditional approach to displaying a key choice to the user, but of course is by no means required.